File: src/main/java/com/github/aivancioglo/resttest/setters/Param.java
package com.github.aivancioglo.resttest.setters;

import com.github.aivancioglo.resttest.http.HTTPRequest;

public class Param<T extends HTTPRequest> implements Setter<T> {
    private String key;
    private Object value;

    public Param(String key, Object value) {
        this.key = key;
        this.value = value;
    }

    /**
     * Update request.
     * @param request actual request.
     */
    @Override
    public void update(T request) {
        request.addParam(key, value);
    }
}

File: src/main/kotlin/com/github/aivancioglo/resttest/setters/Param.kt
package com.github.aivancioglo.resttest.setters

import com.github.aivancioglo.resttest.http.HTTPRequest

/**
 * Param setter.
 *
 * @constructor is creating variables with requestSpecification param key and value.
 * @param key of your param.
 * @param value of your param.
 */
class Param<in T : HTTPRequest<*>>(private val key: String, private val value: Any) : Setter<T> {

    /**
     * Setting param to your requestSpecification.
     *
     * @param request that will be updated.
     */
    override fun update(request: T) {
        request.requestSpecification.param(key, value)
    }
}